### PR TITLE
KTOR-3462 KTOR-3394 Move modules

### PR DESCRIPTION
**Pull request by rsinukov**
* CIO server related code from http-cio to server-cio
* WS from http-cio to websockets
* Rename ios engine to darwin
* Delete ktor-legacy

### CODE REVIEW

File: gradle/compatibility.gradle [hunk 1]
```diff
@@ -16,7 +16,7 @@ apiValidation {
     if (rootProject.ext.native_targets_enabled) {
         excludeList += [
             'ktor-client-curl',
-            'ktor-client-ios',
+            'ktor-client-darwin',
```
Comment (e5l) [inline gradle/compatibility.gradle:19]: We also should add `ktor-client-ios`

File: ktor-http/ktor-http-cio/common/src/io/ktor/http/cio/internals/Chars.kt [hunk 2]
```diff
@@ -66,7 +66,7 @@ internal fun CharSequence.parseHexLong(): Long {
     return result
 }
 
-internal fun CharSequence.parseDecLong(): Long {
+public fun CharSequence.parseDecLong(): Long {
```
Comment (e5l) [inline ktor-http/ktor-http-cio/common/src/io/ktor/http/cio/internals/Chars.kt:69]: KDoc should be added for the new public API. @andreyaksenov could you check the KDocs?

**Review by e5l — CHANGES_REQUESTED** [review]
Could you also add an alias for engine name(`Darwin`)?

File: ktor-client/ktor-client-cio/jvm/src/io/ktor/client/plugins/websocket/buildersCio.kt [hunk 3]
```diff
@@ -8,7 +8,7 @@ import io.ktor.client.*
 import io.ktor.client.call.*
 import io.ktor.client.request.*
 import io.ktor.http.*
-import io.ktor.http.cio.websocket.*
+import io.ktor.websocket.*
 
 /**
  * Create raw [ClientWebSocketSession]: no ping-pong and other service messages are used.
```
Comment (andreyaksenov) [inline ktor-client/ktor-client-cio/jvm/src/io/ktor/client/plugins/websocket/buildersCio.kt:14]: Create -> Creates

File: ktor-client/ktor-client-core/common/src/io/ktor/client/plugins/websocket/builders.kt [hunk 4]
```diff
@@ -8,7 +8,7 @@ import io.ktor.client.*
 import io.ktor.client.call.*
 import io.ktor.client.request.*
 import io.ktor.http.*
-import io.ktor.http.cio.websocket.*
+import io.ktor.websocket.*
 
 /**
  * Install [WebSockets] plugin using the [config] as configuration.
```
Comment (andreyaksenov) [inline ktor-client/ktor-client-core/common/src/io/ktor/client/plugins/websocket/builders.kt:14]: > Installs the [WebSockets] plugin ...

File: ktor-http/ktor-http-cio/common/src/io/ktor/http/cio/internals/MutableRange.kt [hunk 5]
```diff
@@ -4,8 +4,6 @@
 
 package io.ktor.http.cio.internals
 
-import io.ktor.util.*
-
 /**
  * Represents a text range with mutable bounds
```
Comment (andreyaksenov) [inline ktor-http/ktor-http-cio/common/src/io/ktor/http/cio/internals/MutableRange.kt:8]: > A text range with mutable bounds

**Review by andreyaksenov — CHANGES_REQUESTED** [review]

**Review by andreyaksenov — APPROVED** [review]

**Review by e5l — APPROVED** [review]
LGTM

**Pull request merged**